File: cocos/renderer/gfx/gfx-framebuffer.ts
import { GFXDevice } from './gfx-device';
import { GFXTextureView } from './gfx-texture-view';
import { GFXRenderPass } from './gfx-render-pass';

export interface GFXFramebufferInfo {
    renderPass: GFXRenderPass;
    colorViews?: GFXTextureView[];
    depthStencilView?: GFXTextureView;
    isOffscreen?: boolean;
};

export abstract class GFXFramebuffer {

    constructor(device: GFXDevice) {
        this._device = device;
    }

    public abstract initialize(info: GFXFramebufferInfo): boolean;
    public abstract destroy(): void;

    protected _device: GFXDevice;
    protected _colorViews: GFXTextureView[] = [];
    protected _depthStencilView: GFXTextureView | null = null;
    protected _isOffscreen: boolean = true;
};

File: native/cocos/renderer/core/gfx/GFXFramebuffer.cc
#include "CoreStd.h"
#include "GFXFramebuffer.h"

namespace cc {
namespace gfx {

GFXFramebuffer::GFXFramebuffer(GFXDevice *device)
: GFXObject(GFXObjectType::FRAMEBUFFER), _device(device) {
}

GFXFramebuffer::~GFXFramebuffer() {
}

} // namespace gfx
} // namespace cc
